daemon/cdi.go

```diff
@@ -18,8 +18,14 @@ type cdiHandler struct {
 
 // RegisterCDIDriver registers the CDI device driver.
 // The driver injects CDI devices into an incoming OCI spec and is called for DeviceRequests associated with CDI devices.
-func RegisterCDIDriver(opts ...cdi.Option) {
-	cache, err := cdi.NewCache(opts...)
+// If the list of CDI spec directories is empty, the driver is not registered.
+func RegisterCDIDriver(cdiSpecDirs ...string) {
+	if len(cdiSpecDirs) == 0 {
+		log.G(context.TODO()).Info("CDI device driver not registered. No CDI spec directories specified")
+		return
```

Comment:
How about registering a spec updater that always returns an error, like when initialization fails? This log message is only visible to the administrator/operator; users would just get a confusing `could not select device driver "cdi"` error message when they try to run a container with a CDI device attached.